File: src/Application Specific/Classes/Entity Collections/StandardSentences/StandardSentenceSQL.cs
using System.Data;

namespace FSM.Entity
{
    namespace StandardSentences
    {
        public class StandardSentenceSQL
        {
            private Sys.Database _database;

            public StandardSentenceSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public DataView GetAll()
            {
                _database.ClearParameter();
                var dt = _database.ExecuteSP_DataTable("StandardSentence_GetAll");
                dt.TableName = Sys.Enums.TableNames.tblStandardSentences.ToString();
                return new DataView(dt);
            }

            public void Insert(StandardSentence oStandardSentence)
            {
                _database.ClearParameter();
                AddSentenceParametersToCommand(ref oStandardSentence);
                _database.ExecuteSP_NO_Return("StandardSentence_Insert");
            }

            public void Update(StandardSentence oStandardSentence)
            {
                _database.ClearParameter();
                _database.AddParameter("@SentenceID", oStandardSentence.SentenceID, true);
                AddSentenceParametersToCommand(ref oStandardSentence);
                _database.ExecuteSP_NO_Return("StandardSentence_Update");
            }

            private void AddSentenceParametersToCommand(ref StandardSentence oStandardSentence)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@Sentence", oStandardSentence.Sentence, true);
                }
            }

            public void Delete(int SentenceID)
            {
                _database.ClearParameter();
                _database.AddParameter("@SentenceID", SentenceID, true);
                _database.ExecuteSP_NO_Return("StandardSentence_Delete");
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */

        }
    }
}

File: src/Application Specific/Classes/Entity Collections/StandardSentences/StandardSentenceSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace StandardSentences

        Public Class StandardSentenceSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function [GetAll]() As DataView
                _database.ClearParameter()
                Dim dt As DataTable = _database.ExecuteSP_DataTable("StandardSentence_GetAll")
                dt.TableName = Entity.Sys.Enums.TableNames.tblStandardSentences.ToString
                Return New DataView(dt)
            End Function

            Public Sub [Insert](ByVal oStandardSentence As StandardSentence)
                _database.ClearParameter()
                AddSentenceParametersToCommand(oStandardSentence)
                _database.ExecuteSP_NO_Return("StandardSentence_Insert")

            End Sub

            Public Sub [Update](ByVal oStandardSentence As StandardSentence)
                _database.ClearParameter()
                _database.AddParameter("@SentenceID", oStandardSentence.SentenceID, True)
                AddSentenceParametersToCommand(oStandardSentence)
                _database.ExecuteSP_NO_Return("StandardSentence_Update")
            End Sub

            Private Sub AddSentenceParametersToCommand(ByRef oStandardSentence As StandardSentence)
                With _database
                    .AddParameter("@Sentence", oStandardSentence.Sentence, True)
                End With
            End Sub

            Public Sub [Delete](ByVal SentenceID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@SentenceID", SentenceID, True)
                _database.ExecuteSP_NO_Return("StandardSentence_Delete")
            End Sub

#End Region


        End Class

    End Namespace

End Namespace